ompi/mca/osc/ucx/osc_ucx_comm.c

```diff
@@ -544,15 +708,53 @@ int ompi_osc_ucx_accumulate(const void *origin_addr, int origin_count,
             return ret;
         }
 
-        ret = opal_common_ucx_wpmem_flush(module->mem, OPAL_COMMON_UCX_SCOPE_EP, target);
+    }
+
+    if (NULL != ucx_req) {
+        // nothing to wait for, mark request as completed
+        ompi_request_complete(&ucx_req->super, true);
+    }
+
+    return end_atomicity(module, target, lock_acquired, free_ptr);
```

Comment:
`free_ptr` is passed to `end_atomicity` because the transfer is overlapped with the release of the atomic lock and only after that the `free_ptr` can actually be free'd.